File: cpp/026.cpp
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        if (nums.empty())
            return 0;
        int cur = 0, size = nums.size();
        for (int i = 1; i < size; ++i) {
            if (nums[cur] != nums[i])
                nums[++cur] = nums[i];
        }
        return cur + 1;
    }
};

File: go/026.go
func removeDuplicates(nums []int) int {
    cur, size := 0, len(nums)
    if size == 0 {
        return 0
    }
    for i := 1; i < size; i++ {
        if nums[cur] != nums[i] {
            cur++
            nums[cur] = nums[i]
        }
    }
    return cur + 1
}
